tests/test_selector_jmespath.py

```diff
@@ -0,0 +1,94 @@
+# -*- coding: utf-8 -*-
+
+import unittest
+
+from parsel import Selector
+
+
+class JMESPathTestCase(unittest.TestCase):
```

Comment:
What's the behavior of `.jmespath(...).re()`, would it make sense to test it?

Reply:
I expect it to fail on non-strings with the current implementation, and that may be best. Otherwise, we could JSON-encode the result first. Testing the scenario (and maybe covering it in the docs somehow?) would be great.

Reply:
I tested the behavior for jmespath with re and overall the approach to JSON encoding the selector's results seemed to be the more straightforward one for giving a complete result at the moment.  This is also easier given the simplified approach I went with for the type argument inside the selector.